lib/rufo/formatter/settings.rb

```diff
@@ -1,93 +1,27 @@
-    parens_in_def options.fetch(:parens_in_def, :yes)
-    double_newline_inside_type options.fetch(:double_newline_inside_type, :dynamic)
-    align_case_when options.fetch(:align_case_when, false)
-    align_chained_calls options.fetch(:align_chained_calls, false)
-    trailing_commas options.fetch(:trailing_commas, :always)
-  end
-
-  def spaces_around_binary(value)
-    @spaces_around_binary = one_dynamic("spaces_around_binary", value)
-  end
-
-  def parens_in_def(value)
-    @parens_in_def = yes_dynamic("parens_in_def", value)
-  end
-
-  def double_newline_inside_type(value)
-    @double_newline_inside_type = no_dynamic("double_newline_inside_type", value)
-  end
-
-  def trailing_commas(value)
-    case value
-    when :always, :never
-      @trailing_commas = value
-    else
-      raise ArgumentError.new("invalid value for trailing_commas: #{value}. Valid values are: :dynamic, :always, :never")
-    end
-  end
-
-  def align_case_when(value)
-    @align_case_when = value
-  end
-
-  def align_chained_calls(value)
-    @align_chained_calls = value
-  end
+  OPTIONS = {
+    spaces_around_binary: [:dynamic, :one],
+    parens_in_def: [:yes, :dynamic],
+    double_newline_inside_type: [:dynamic, :no],
+    align_case_when: [false, true],
+    align_chained_calls: [false, true],
+    trailing_commas: [:always, :never],
+  }
 
-  def dynamic_always_never(name, value)
-    case value
-    when :dynamic, :always, :never
-      value
-    else
-      raise ArgumentError.new("invalid value for #{name}: #{value}. Valid values are: :dynamic, :always, :never")
-    end
-  end
-
-  def dynamic_always_never_match(name, value)
-    case value
-    when :dynamic, :always, :never, :match
-      value
-    else
-      raise ArgumentError.new("invalid value for #{name}: #{value}. Valid values are: :dynamic, :always, :never, :match")
-    end
-  end
-
-  def one_dynamic(name, value)
-    case value
-    when :one, :dynamic
-      value
-    else
-      raise ArgumentError.new("invalid value for #{name}: #{value}. Valid values are: :one, :dynamic")
-    end
-  end
-
-  def no_dynamic(name, value)
-    case value
-    when :no, :dynamic
-      value
-    else
-      raise ArgumentError.new("invalid value for #{name}: #{value}. Valid values are: :no, :dynamic")
-    end
-  end
-
-  def one_no_dynamic(name, value)
-    case value
-    when :no, :one, :dynamic
-      value
-    else
-      raise ArgumentError.new("invalid value for #{name}: #{value}. Valid values are: :no, :one, :dynamic")
+  def init_settings(options)
+    OPTIONS.each do |name, valid_options|
+      default = valid_options.first
+      value = options.fetch(name, default)
+      unless valid_options.include?(value)
+        $stderr.puts "Invalid value for #{name}: #{value.inspect}. Valid " \
+                     "values are: #{valid_options.map(&:inspect).join(', ')}"
+        value = default
+      end
+      self.instance_variable_set("@#{name}", value)
```

Comment:
Relying on instance variables seems error prone to me, leaves us open to typoing the setting entirely in the formatter and not realising it. What about adding
```ruby
attr_accessor *OPTIONS.keys
```
... to the class, and replacing this with
```
public_send(name, value)
```
and then referring to these settings by their methods rather than instance variables in the formatter?

Reply:
Side note, this isn't you, but having `rufo/formatter/settings.rb` reopen an existing class and add methods to it had me very confused. How about this is instead a `Rufo::Settings` module that is mixed into `Rufo::Formatter` explicitly?